### DISCUSSION TITLE
Apdex plugin

### DISCUSSION THREAD

**Discussion by Mathdon** [artilleryio/artillery#2209] [Artillery]
Hi

I am using the canary deployment of artillery 2.0.0-37-9c51118

And using the apdex and ensure plugin.

I add a check in the ensure plugin for apdex.frustrated

`  plugins:
    apdex:
      threshold: 500
    ensure:
      thresholds:
        - "http.response_time.p99": 2000
        - "vusers.failed": 1
        - "apdex.frustrated": 99`

This works fine when there are frustrated metrics, but when no requests were frustrated the check fails as no metrics were reported.
Is there a way around this?

Would it also be possible to pass in the frustrated threshold? I believe the frustrated threshold is 4x greater than the threshold set.
Could we pass in a frustrated threshold so we are able to configure this from the test script?

Thanks

**Comment by hassy** [comment]
hi @Mathdon 👋 I think this is fixed on `main` now, by #2210 and #2213.

**Reply by InesNi** [reply]
That's correct - apologies @Mathdon, I meant to mention here that this should be fixed, and you can use our canary version `artillery@2.0.0-37-77ea3e9` to unblock yourself for now. It will also be released soon in Artillery 2.0.0-38.

Thanks again for reporting it!

---

As for your other question:
>Would it also be possible to pass in the frustrated threshold?

If I understand correctly, you're asking if you can control the frustrated threshold specifically from the Apdex plugin? You can only control the overall threshold, and then the `tolerated`, `frustrated` etc values are calculated by the usual apdex methods against that threshold, which can't be customised. You can read more about it here: https://en.wikipedia.org/wiki/Apdex

**Reply by Mathdon** [reply]
Thank you for your help 😀

**Reply by bernardobridge** [reply]
These fixes are now released in [Artillery v2.0.0-38](https://github.com/artilleryio/artillery/releases/tag/v2.0.0-38).
